### DISCUSSION TITLE
i18n-js v4.0.0-alpha.8 has been released

### DISCUSSION THREAD

**Discussion by fnando** [fnando/i18n-js#612] [Announcements]
Hey!

A new alpha release is here: v4.0.0-alpha.8! This is a *breaking change* release.

This version matches Rails API, introducing the same method names and options.

- `I18n#numberToCurrency`
- `I18n#numberToDelimited`
- `I18n#numberToRounded`
- `I18n#numberToHuman`
- `I18n#numberToHumanSize`
- `I18n#numberToPercentage`

There's also a new way of tracking changes to I18n's store.

```js
const i18n = new I18n();

// I18n#version is incremented every time I18n#locale and I18n#defaultLocale changes, 
// I18n#update or I18n#store is called.
i18n.onChange(i18n => console.log("current version:", i18n.version));
```

**Comment by asadakbarml** [comment]
Does the I18n.store cache the translations? When is the cache rewritten?

**Comment by dialex** [comment]
Do you have a list of breaking changes? These properties `fallbacks`, `translations`, `locale` no longer exist in `i18n-js/typings/index`.

**Reply by dialex** [reply]
@fnando So I found [this migration guide](https://github.com/fnando/i18n-js/blob/main/MIGRATING_FROM_V3_TO_V4.md), but I still have doubts.

> `fallbacks` was removed without an equivalent
FIne, I also removed it from our code ✅ 

> `i18n.translations = { en };`
How do we migrate this ❓ 

> `i18n.locale = locale || 'en';`
How do we migrate this ❓

**Reply by dialex** [reply]
We solved it. This is how we do it now:
```js
export const i18n = new I18n(
  { en },
  { locale: locale ?? 'en', enableFallback: true },
);
```
